gwcs/geometry.py

```diff
@@ -74,68 +76,134 @@ class SphericalToCartesian(Model):
     n_inputs = 2
     n_outputs = 3
 
-    def __init__(self, **kwargs):
+    def __init__(self, wrap_lon_at=360, **kwargs):
+        """
+        Parameters
+        ----------
+        wrap_lon_at : {360, 180}, optional
+            An **integer number** that specifies the range of the longitude
+            (azimuthal) angle. When ``wrap_lon_at`` is 180, the longitude angle
+            will have a range of ``[-180, 180)`` and when ``wrap_lon_at``
+            is 360 (default), the longitude angle will have a range of
+            ``[0, 360)``.
+
+        """
         super().__init__(**kwargs)
-        self.inputs = ('phi', 'theta')
+        self.inputs = ('lon', 'lat')
         self.outputs = ('x', 'y', 'z')
+        self.wrap_lon_at = wrap_lon_at
 
-    @staticmethod
-    def evaluate(phi, theta):
-        if isinstance(phi, u.Quantity) != isinstance(theta, u.Quantity):
+    @property
+    def wrap_lon_at(self):
+        """ An **integer number** that specifies the range of the longitude
+        (azimuthal) angle.
+
+        Allowed values are 180 and 360. When ``wrap_lon_at``
+        is 180, the longitude angle will have a range of ``[-180, 180)`` and
+        when ``wrap_lon_at`` is 360 (default), the longitude angle will have a
+        range of ``[0, 360)``.
+
+        """
+        return self._wrap_lon_at
+
+    @wrap_lon_at.setter
+    def wrap_lon_at(self, wrap_angle):
+        if not (isinstance(wrap_angle, numbers.Integral) and wrap_angle in [180, 360]):
+            raise ValueError("'wrap_lon_at' must be an integer number: 180 or 360")
+        self._wrap_lon_at = wrap_angle
+
+    def evaluate(self, lon, lat):
+        if isinstance(lon, u.Quantity) != isinstance(lat, u.Quantity):
             raise TypeError("All arguments must be of the same type "
                             "(i.e., quantity or not).")
 
-        phi = np.deg2rad(phi)
-        theta = np.deg2rad(theta)
+        lon = np.deg2rad(lon)
+        lat = np.deg2rad(lat)
 
-        cs = np.cos(theta)
-        x = cs * np.cos(phi)
-        y = cs * np.sin(phi)
-        z = np.sin(theta)
+        cs = np.cos(lat)
+        x = cs * np.cos(lon)
+        y = cs * np.sin(lon)
+        z = np.sin(lat)
 
         return x, y, z
 
     def inverse(self):
-        return CartesianToSpherical()
+        return CartesianToSpherical(wrap_lon_at=self._wrap_lon_at)
 
     @property
     def input_units(self):
-        return {'phi': u.deg, 'theta': u.deg}
+        return {'lon': u.deg}
```

Comment:
Is it intentional that we're dropping `lat` from this dictionary?

Reply:
Good catch! This was accidental.